src/Bootstraps/ConfigBootstrap.php

```diff
@@ -0,0 +1,31 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Tempest\Bootstraps;
+
+use Tempest\AppConfig;
+use Tempest\Container\Container;
+use function Tempest\path;
+
+final readonly class ConfigBootstrap implements Bootstrap
+{
+    public function __construct(
+        private Container $container
+    ) {}
+
+    #[\Override]
+    public function boot(): void
+    {
+        // Scan for package config files
+        foreach ($this->container->get(AppConfig::class)->discoveryLocations as $package) {
```

Comment:
Why not just making AppConfig a dependency in the constructor and allow the container to resolve it when getting the class?

Reply:
i'm not sure i follow. `AppConfig` is in the Container (a singleton via `config()`), so why make it a separate dependency?

Reply:
Since you are now using the `get` method on the container to create the bootstrap class, the container will provide any dependencies in the `__construct` method. `AppConfig` could simply move to the construct as a dependency. 🙂

Reply:
Here's what i see happening in Spiral:

The Kernel is constructed by passing an _array_ of named directories:

[Tutorial app example](https://github.com/spiral/app/blob/master/app.php#L20)
[Method declaration](https://github.com/spiral/framework/blob/master/src/Boot/src/AbstractKernel.php#L109)

The array becomes a `Directories` object in the Container:
[Kernel constructor](https://github.com/spiral/framework/blob/master/src/Boot/src/AbstractKernel.php#L85)

It may be worth noting the Kernel builds the Container in `Kernel::create()` and passes it to the constuctor... i guess that permits other approaches to construct the Kernel with a different Container implementation, where Tempest's Kernel wants it done with the Tempest Container.

For future consideration, we may want to refactor to permit other Containers, but not today.

---

If there's no opposition, i'm going to add a similar `Directories` class that the Container will be able to resolve, and take care of this exposure issue.

Reply:
So in reviewing, I actually think Spiral is a pretty poor example of a bootstrapper implementation.

Laravel, I think, gives a better perspective into what a bootstrapper _should_ be doing. See [here](https://github.com/laravel/framework/blob/698c74b8465daa518cf3954eea133e4e6e3f47d7/src/Illuminate/Foundation/Http/Kernel.php#L42) for example.

A bootstrapper isn't really getting hands on with the application. It is getting and providing the data necessary to help setup the kernel so that the actual application can load. Much like a computers boot process.

To me, there are a few things that are concerning to me about this PR:

1. It seems to have taken a turn away from kernel bootstrapping and toward setting up dependencies in the container. We basically already have this in initializers, so I'm not entirely sure what we are gaining.

2. The scope of the PR keeps changing/getting added to.

I don't say that to discourage you from going down this path, but only to encourage you to target one specific problem to tackle here so that we can merge this specific PR quickly and look to tackle other issues in future PRs. It would be helpful to have a clear argument for what the issue is that we are trying to solve, and how bootloaders/bootstrappers are going to do that for us.

Reply:
Oops, i meant that Directories comment for the discussion on `__construct()` ...

i apologize for responding to the wrong thread here... i'll copy the comments up to that thread, so hopefully that will get it back on track.

---

Focusing on _this_ thread originally being about passing AppConfig and Container separately to the ConfigBootstrap: the Container has AppConfig already, and we're letting DI pass the Container to the Bootstrap, so all the resources ConfigBootstrap needs are available in one parameter. i'm still not clear what advantage there is to separating them.